pkg/azure/api/api.go

```diff
@@ -189,6 +190,10 @@ func (c *Client) vmssNetworkInterfaces(ctx context.Context) ([]network.Interface
 		result2, err2 := c.interfaces.ListVirtualMachineScaleSetNetworkInterfacesComplete(ctx, c.resourceGroup, *scaleset.Name)
 		c.metricsAPI.ObserveAPICall("Interfaces.ListVirtualMachineScaleSetNetworkInterfacesComplete", deriveStatus(err2), sinceStart.Seconds())
 		if err2 != nil {
+			// For scale set without any instances API will return not found. Then it can be skipped.
+			if v, ok := err2.(autorest.DetailedError); ok && v.StatusCode == http.StatusNotFound {
```

Comment:
Makes sense, I did a bit more digging. Strangely, it seems like the Azure API only returns NotFound when the vmss was created as part of a AKS node-group, otherwise it returns a 200 with any empty list.

Could we document this idiosyncrasy in the comment for the next person looking at this.

Reply:
sure

Reply:
Done